server/executor/default_poller_executor.go

```diff
@@ -196,15 +201,15 @@ func (pe DefaultPollerExecutor) testConnection(ctx context.Context, traceDB trac
 	return nil
 }
 
-func (pe DefaultPollerExecutor) donePollingTraces(job *Job, traceDB tracedb.TraceDB, trace model.Trace) (bool, string) {
+func (pe DefaultPollerExecutor) donePollingTraces(job *Job, traceDB tracedb.TraceDB, trace traces.Trace) (bool, string) {
 	if !traceDB.ShouldRetry() {
 		return true, "TraceDB is not retryable"
 	}
 
 	maxTracePollRetry := job.PollingProfile.Periodic.MaxTracePollRetry()
 	// we're done if we have the same amount of spans after polling or `maxTracePollRetry` times
 	log.Printf("[PollerExecutor] Test %s Run %d: Job count %d, max retries: %d", job.Test.ID, job.Run.ID, job.EnqueueCount(), maxTracePollRetry)
-	if job.EnqueueCount() == maxTracePollRetry {
+	if job.EnqueueCount() >= maxTracePollRetry {
```

Comment:
there was a weird bug where a job got incorrectly requeued after hitting the max retries limit. Since the new `EnqueueConunt` was bigger than `maxTracePollRetry` and the check was for equality, it got forever stuck in a polling loop